scripts/uberenv/packages/axom/package.py

```diff
@@ -314,83 +340,107 @@ def hostconfig(self, spec, prefix):
+                python_bin_dir = get_spec_path(spec, "python",
+                                               path_replacements,
+                                               use_bin=True)
+                cfg.write(cmake_cache_entry("SPHINX_EXECUTABLE",
+                                            pjoin(python_bin_dir,
+                                                  "sphinx-build")))
         else:
             cfg.write(cmake_cache_option("ENABLE_DOCS", False))
 
         if "py-shroud" in spec:
-            python_bin_dir = get_spec_path(spec, "python", path_replacements, use_bin=True)
-            cfg.write(cmake_cache_entry("SHROUD_EXECUTABLE", pjoin(python_bin_dir, "shroud")))
+            shroud_bin_dir = get_spec_path(spec, "shroud",
+                                           path_replacements, use_bin=True)
+            cfg.write(cmake_cache_entry("SHROUD_EXECUTABLE",
+                                        pjoin(shroud_bin_dir, "shroud")))
 
         if "uncrustify" in spec:
-            uncrustify_bin_dir = get_spec_path(spec, "uncrustify", path_replacements, use_bin=True)
-            cfg.write(cmake_cache_entry("UNCRUSTIFY_EXECUTABLE", pjoin(uncrustify_bin_dir, "uncrustify")))
+            uncrustify_bin_dir = get_spec_path(spec, "uncrustify",
+                                               path_replacements,
+                                               use_bin=True)
+            cfg.write(cmake_cache_entry("UNCRUSTIFY_EXECUTABLE",
+                                        pjoin(uncrustify_bin_dir,
+                                              "uncrustify")))
 
         if "cppcheck" in spec:
-            cppcheck_bin_dir = get_spec_path(spec, "cppcheck", path_replacements, use_bin=True)
-            cfg.write(cmake_cache_entry("CPPCHECK_EXECUTABLE", pjoin(cppcheck_bin_dir, "cppcheck")))
-
+            cppcheck_bin_dir = get_spec_path(spec, "cppcheck",
+                                             path_replacements, use_bin=True)
+            cfg.write(cmake_cache_entry("CPPCHECK_EXECUTABLE",
+                                        pjoin(cppcheck_bin_dir, "cppcheck")))
 
         ##################################
         # Other machine specifics
         ##################################
 
-        cfg.write("#------------------{}\n".format("-"*60))
+        cfg.write("#------------------{0}\n".format("-" * 60))
         cfg.write("# Other machine specifics\n")
-        cfg.write("#------------------{}\n\n".format("-"*60))
+        cfg.write("#------------------{0}\n\n".format("-" * 60))
 
         # OpenMP
         if "+openmp" in spec:
             cfg.write(cmake_cache_option("ENABLE_OPENMP", True))
+        else:
+            cfg.write(cmake_cache_option("ENABLE_OPENMP", False))
 
         # Enable death tests
-        if on_blueos and "+cuda" in spec:
+        if spec.satisfies('target=ppc64le:') and "+cuda" in spec:
```

Comment:
I'm assuming they didn't like using the `SYS_TYPE` to determine which platform we're on.
On the other hand, it still seems cleaner to extract the architecture-based logic into its own variable.

Could we define `on_blueos = spec.satisfies('target=ppc64le:')` ?

Reply:
kinda defeats the whole purpose of going with the architecture.  For example, this works on summit which is not blueos apparently.